docs/usage.rst

```diff
@@ -588,64 +588,17 @@ Run Twisted in a separate thread
 --------------------------------
 
 To run Dispersy in a separate thread you have to run twisted in a separate thread. This can be done by overriding the
-start and stop functions of twisted as can be seen below. This code is best placed in a separate file. The import to
-the start_reactor() function should be the first import to the twisted library in your project otherwise the twisted
-library is already initialised and will be running on the same thread.
-
-.. code-block:: python
-
-    from twisted.internet import reactor
-    from twisted.python import log
-
-    _twisted_thread = None
-
-
-    def start_reactor():
-        """
-        Start the Twisted reactor in a separate thread, if not already done and return the reactor.
-        """
-        global _twisted_thread
-        if not _twisted_thread:
-            from threading import Thread
-
-            def _start_reactor():
-                reactor.suggestThreadPoolSize(1)
-                reactor.run(installSignalHandlers=False)
-
-            _twisted_thread = Thread(target=_start_reactor, name="Twisted")
-            _twisted_thread.setDaemon(True)
-            _twisted_thread.start()
-
-            def hook_observer():
-                observer = log.PythonLoggingObserver()
-                observer.start()
-                import logging
-                log.msg("PythonLoggingObserver hooked up", logLevel=logging.DEBUG)
-
-            reactor.callFromThread(hook_observer)
-
-        return reactor, _twisted_thread
-
-
-    def stop_reactor():
-        """
-        Stop the reactor and join the reactor thread until it stops.
-        """
-
-        global _twisted_thread
-
-        def _stop_reactor():
-            reactor.stop()
-
-        reactor.callFromThread(_stop_reactor)
-        _twisted_thread.join()
-        for p in reactor.getDelayedCalls():
-            if p.active():
-                p.cancel()
-        _twisted_thread = None
+start and stop functions of twisted as can be seen in this repository from Tribler
```

Comment:
I wouldn't recommend people to run twisted on a separate thread, it's just looking for trouble (we do it on Tribler because of WX taking over the main thread, and even that it's going away soon). 

Your dispersy based program should be async and use twisted, even better if it's a twistd plugin.